Add API that allows passing CKM_MECHANISM parameters

selvanair opened the pull request:
Two commits as part 1 of PR #31 replacement

- Add some defines for PSS
- Add extended API for sign and decrypt

PS: feel free to suggest changes to function names, formatting and other nits..

alonbl commented on include/pkcs11-helper-1.0/pkcs11.h:
```diff
@@ -162,6 +162,10 @@ extern "C" {
 #define min_key_size ulMinKeySize
```
I synced the pkcs11.h with opensc master, please rebase, you can drop this.

alonbl commented on lib/pkcs11h-certificate.c:
```diff
@@ -786,9 +786,6 @@ __pkcs11h_certificate_doPrivateOperation (
-	CK_MECHANISM mech = {
-		mech_type, NULL, 0
-	};
 
```
can you please add assert with NULL for all these?

can you please add a debug with mech_type after the asserts so that we have some visibility of what is going on?

alonbl commented on lib/pkcs11h-certificate.c:
```diff
@@ -1316,9 +1333,9 @@ pkcs11h_certificate_sign (
+		"PKCS#11: pkcs11h_certificate_sign_ex entry certificate=%p, mech_type=%ld, source=%p, source_size="P_Z", target=%p, *p_target_size="P_Z"",
 		(void *)certificate,
-		mech_type,
+		mech->mechanism,
```
once you add the assert above, you can do this :)

selvanair replied:
Good catch!

alonbl reviewed (commented):
Thanks! a minor comment.

selvanair commented:
Force-pushed after rebase and edits:
- Removed the first commit obsoleted by the opensc sync
- Added the asserts, included mech in debug print and added the new _ex functions to exports

alonbl commented:
Thanks! merged with trivial changes.

selvanair commented:
Thanks. yeah, I misunderstood your comment about debug prints..
